File: src/python/src/transport/poly_bus_not_started_error.py
"""
PolyBus error indicating that the bus has not been started.
"""

from ..poly_bus_error import PolyBusError


class PolyBusNotStartedError(PolyBusError):
    """A PolyBus error indicating that the bus has not been started."""

    def __init__(self) -> None:
        super().__init__(
            1, "PolyBus has not been started. Please call IPolyBus.start() before using the bus."
        )

File: src/typescript/src/transport/poly-bus-not-started-error.ts
import { PolyBusError } from '../poly-bus-error';

/**
 * A PolyBus error indicating that the bus has not been started.
 */
export class PolyBusNotStartedError extends PolyBusError {
  /**
   * Creates a new PolyBusNotStartedError instance.
   */
  constructor() {
    super(1, 'PolyBus has not been started. Please call IPolyBus.start() before using the bus.');
    this.name = 'PolyBusNotStartedError';

    // Maintains proper stack trace for where our error was thrown (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, PolyBusNotStartedError);
    }
  }
}
